src/Hedgehog/Linq/Property.fs

```diff
@@ -5,6 +5,16 @@ namespace Hedgehog.Linq
 open System
 open System.Runtime.CompilerServices
 open Hedgehog
+open System.Runtime.InteropServices
+
+module internal Build =
+    let config tests shrinkLimit =
+        PropertyConfig.defaultConfig
+        |> PropertyConfig.withTestCount (tests |> Option.defaultValue PropertyConfig.defaultConfig.TestCount)
+        |> fun config ->
+            match shrinkLimit with
+            | Some shrinkLimit -> config |> PropertyConfig.withShrinkLimit shrinkLimit
+            | None -> config
```

Comment:
We should probably expose a fluent builder API for consumption in other .NET languages, something like this:

```csharp
var config = PropertyConfig.Default
    .WithShrinkLimit(500)
    .WithTests(100);
```

Then these methods can just accept a `PropertyConfig` parameter, like the underlying functions. The main reason to do it this way is if we add new configuration parameters later, then we just have to add that to the builder API.

Reply:
I added support for both. The look of line 91 in this screenshot doesn't look great to me, but perhaps there's a common config a consumer wants to use everywhere. I want to keep the overloads because they're what C#/VBers are used to, and it feels weird to force them to use the config. We can add more optional overloads later, and since they're optional it won't break anything.

![image](https://user-images.githubusercontent.com/1278803/105780919-2d226180-5f37-11eb-9935-a2cf046f9201.png)

Reply:
This can appear a little clunky, but I also wouldn't write it this way. Maybe this is a preference thing, but I would probably write that as:

```csharp
// Define an acceptable configuration as a baseline for most tests
var config = PropertyConfig.Default;

// Override when necessary locally
property.Check(config.WithTestCount(20));
```

The parameterized approach is less strict, to be sure. The question I want to raise is: Is that a good thing?

I think we should decide on a single approach to keep the API clean. There's nothing worse when using a library than having too many ways to do something.

Subtle note about the code above: I've chosen the name `PropertyConfig.Default` as this is something of an unwritten standard in C#.

```csharp
var x = System.Collections.Generic.EqualityComparer<string>.Default;
var y = System.Threading.Tasks.TaskScheduler.Default;
```

I'm not bothered if we want to go with `DefaultConfig`, as it matches the underlying code better though.

Reply:
`PropertyConfig.DefaultConfig` is verbose.  No need to repeate `config`.

Reply:
In your example, `config` is scoped to a single method. I believe most testing frameworks (xUnit, nUnit, MSTest) have a separate method per test. So in order to span multiple tests, the config will need to be a class level member. It is likely that users will want their PropertyConfigs to span multiple classes though, so we're back to a static class (e.g. `TestHelper`) that sets the standard `PropertyConfig` (e.g. `TestHelper.DefaultConfig`). So to override a local test, it would still look similar to what I have in the screenshot.

I personally feel like two ways of doing things is fine. My personal style is to make it as easy as possible for the end-user to consume.

Regarding `PropertyConfig.Default` I agree and will push an update momentarily.

Reply:
> In your example, `config` is scoped to a single method. I believe most testing frameworks (xUnit, nUnit, MSTest) have a separate method per test. So in order to span multiple tests, the config will need to be a class level member. It is likely that users will want their PropertyConfigs to span multiple classes though, so we're back to a static class (e.g. `TestHelper`) that sets the standard `PropertyConfig` (e.g. `TestHelper.DefaultConfig`). So to override a local test, it would still look similar to what I have in the screenshot.

I think this might be handled better by integration packages (`Hedgehog.MSTest`, `Hedgehog.Xunit`, etc). Since hedgehog itself isn't a test runner, maybe in a package like that we can make the integration nicer. Your concerns are valid though, just trying to decide what's best for this project.

Maybe we could punt for now and add some of these things later, as they wouldn't be breaking. I think we're going to have a constant struggle of "how do we make this idiomatic in C#/VB?" and we definitely don't need to settle that to get these changes merged, as they are very nice to have.

Reply:
>I think this might be handled better by integration packages (Hedgehog.MSTest, Hedgehog.Xunit, etc)

Sorry, I don't think I follow. I believe you're suggesting that we move the overloads with optional arguments to another integration package like Hedgehog.Xunit. However, the following overload (and its friends) don't have anything specific to xUnit, MSTest, etc: 

```fsharp
    static member Check
        (   property : Property<bool>,
            [<Optional; DefaultParameterValue null>] ?testLimit   : int<tests>,
            [<Optional; DefaultParameterValue null>] ?shrinkLimit : int<shrinks>
        ) : unit =
        Property.checkBoolWith (Build.config testLimit shrinkLimit) property
```

If we were to go through with this, we would need to make an integration package for every testing framework... with code that isn't specific to any testing framework. It would be a literal copy/paste from framework to framework. Also, this code is (probably) only going to be used by C#ers and VBers, and Hedgehog.Xunit is language agnostic. (For example I use F# with xUnit.) I believe these overloads belong in Hedgehog.LINQ because overloads are what C#ers and VBers are used to.

I'll remove these overloads if you want - just thought I'd respond to that sentence.

Reply:
> Sorry, I don't think I follow.

@dharmaturtle I just mean configuration management for running tests might be better suited for a specific test runner integration, since that was a concern raised in your previous comment. I'm not saying this with certainty since I haven't looked to see how configurable these test runners are.

Reply:
Ah, I see what you mean. I agree that configuration management for specific testing frameworks should be relegated to other packages. I was just listing them as examples of testing frameworks that separate tests as methods of a class, which will necessitate code that looks like

```c#
property.Check(TestHelper.StandardConfig.WithTestCount(20));
```

vs

```c#
property.Check(20);
//or
property.Check(tests:20);
```